runtime/domain.c

```diff
@@ -272,10 +243,18 @@ static void create_domain(uintnat initial_minor_heap_wsize) {
       goto init_major_gc_failure;
     }
 
-    if(caml_reallocate_minor_heap(initial_minor_heap_wsize) < 0) {
-      goto reallocate_minor_heap_failure;
+    /* setting young_limit and young_ptr to minor_heaps_base
+       to trigger minor_heaps reallocation on GC poll */
+    domain_state->young_start = (char*)caml_global_minor_heap_start;
+    domain_state->young_end = (char*)caml_global_minor_heap_start;
+
+    young_limit = atomic_load_acq((atomic_uintnat*)&domain_state->young_limit);
+    if( young_limit != INTERRUPT_MAGIC ) {
+      atomic_compare_exchange_strong((atomic_uintnat*)&domain_state->young_limit, &young_limit, caml_global_minor_heap_start);
```

Comment:
I think this snippet to update the `young_limit` should use `caml_update_young_limit`